---
title: "microbiome-demo"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{microbiome-demo}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(alto)
library(dplyr)
library(magrittr)
library(ggplot2)
library(tibble)
library(stringr)
library(tidyr)
```


This document showcases the `alto` package functions applied to vaginal microbiome data.

## Loading the data

We use the data published with the article ["Replication and refinement of a vaginal microbial signature of preterm birth in two racially distinct cohorts of US women"](https://www.pnas.org/content/114/37/9966.short) by Calahan et al., 2017, PNAS.

A .zip file containing the related data can be downloaded from the [Stanford data catalog](https://stacks.stanford.edu/file/druid:yb681vm1809/RepRefine_Scripts.tar.gz).

For convenience, the data file (i.e. the tables contained in the `processed.rda` file) has been attached to the `alto` package as `vm_data`. For details, type `?vm_data`.

```{r show-names-of-provided-data}

names(vm_data)

```

First, we do some pre-processing of the data. Specifically, we change the `colnames` to replace ASV sequences by human-friendly names built from the taxonomy table.

```{r human-friendly-ASV-names}

# we create the human-friendly ASV names 
# by concatenating Genus and Species and an identifying number 
# as a given species can be represented by several ASVs
vm_data$taxonomy <- 
  vm_data$taxonomy %>% 
  as.data.frame() %>% 
  group_by(Genus, Species) %>% 
  mutate(
    ASV_name = 
      paste0(
        ifelse(is.na(Genus),"[unknown Genus]", Genus), " ",
        ifelse(is.na(Species),"[unknown Species]", Species), " ",
        row_number()
        )
    ) %>% 
  ungroup()

# then, we replace the colnames of the count matrix by these new names. 
# Note that the tax table rows are ordered as the count matrix columns,
# allowing us to assign without matching first.
colnames(vm_data$counts) = vm_data$taxonomy$ASV_name

```



## Using `alto` on these data

Our first step is to run the lda models for varying number of topics, i.e. from 2 to 15 topics.
For this, we use the `run_lda_models` function. This can take a while on a personal computer. 
For example, it takes about x minutes on the authors' laptops.

```{r run-lda-models}

lda_varying_params_lists = list()
for (k in 2:12) {lda_varying_params_lists[[paste0("k",k)]] = list(k = k)}

lda_models = 
  run_lda_models(
    data = vm_data$counts[1:100,1:50], # change for the full data set when full workflow is ready
    lda_varying_params_lists = lda_varying_params_lists,
    lda_fixed_params_list = list(method = "VEM"),
    verbose = TRUE
  )

```

We can now align the topics from each consecutive models:


```{r align-topics}

aligned_topics = 
  align_topics(
    models = lda_models, 
    comparison = "consecutive", 
    method = "product") # transport product


aligned_topics@weights %>% head()

```

And visualize the alignment: 

```{r align-topics-viz}

plot_alignment(aligned_topics)

```


## Diagnostics

```{r key-number-of-topics}

compute_number_of_key_topics(weights = aligned_topics@weights, plot = TRUE)

```


```{r stability}

stability = 
  compute_stability_along_branches(
    aligned_topics@weights
  )

ggplot(stability, aes(x = m, y = branch, fill = stability)) +
  geom_tile() +
  scale_fill_gradient(low = "red", high = "blue")

```

```{r refinement}

refinement <- 
  compute_refinement_scores(
    aligned_topics@weights
  )

ggplot(refinement, 
       aes(x = m, y = k_LDA, fill = refinement_score)) +
  geom_tile()

```

