---
title: "microbiome-demo"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{microbiome-demo}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  eval = FALSE
)
```

```{r setup}
library(alto)
library(dplyr)
library(magrittr)
library(ggplot2)
library(tibble)
library(stringr)
library(tidyr)
```


This document showcases the `alto` package functions applied to vaginal microbiome data.

## Loading the data

We use the data published with the article ["Replication and refinement of a vaginal microbial signature of preterm birth in two racially distinct cohorts of US women"](https://www.pnas.org/content/114/37/9966.short) by Calahan et al., 2017, PNAS.

A .zip file containing the related data can be downloaded from the [Stanford data catalog](https://stacks.stanford.edu/file/druid:yb681vm1809/RepRefine_Scripts.tar.gz).

For convenience, the data file (i.e. the tables contained in the `processed.rda` file) has been attached to the `alto` package as `vm_data`. For details, type `?vm_data`.

```{r show-names-of-provided-data}
names(vm_data)

```

First, we do some pre-processing of the data. Specifically, we change the `colnames` to replace ASV sequences by human-friendly names built from the taxonomy table.

```{r human-friendly-ASV-names}

# we create the human-friendly ASV names
# by concatenating Genus and Species and an identifying number
# as a given species can be represented by several ASVs
vm_data$taxonomy <-
  vm_data$taxonomy %>%
  as.data.frame() %>%
  group_by(Genus, Species) %>%
  mutate(
    ASV_name =
      paste0(
        ifelse(is.na(Genus),"[unknown Genus]", Genus), " ",
        ifelse(is.na(Species),"[unknown Species]", Species), " ",
        row_number()
        )
    ) %>%
  ungroup()

# then, we replace the colnames of the count matrix by these new names.
# Note that the tax table rows are ordered as the count matrix columns,
# allowing us to assign without matching first.
colnames(vm_data$counts) <-  vm_data$taxonomy$ASV_name

```



## Using `alto` on these data

Our first step is to run the lda models for varying number of topics, i.e. from 2 to 15 topics.
For this, we use the `run_lda_models` function. This can take a while on a personal computer.
For example, it takes about x minutes on the authors' laptops.

```{r run-lda-models}
lda_varying_params_lists <-  list()
for (k in 1:20) {lda_varying_params_lists[[paste0("k",k)]] <- list(k = k)}

lda_models  <- 
  run_lda_models(
    data = vm_data$counts[,1:100], # change for the full data set when full workflow is ready
    lda_varying_params_lists = lda_varying_params_lists,
    lda_fixed_params_list = list(method = "VEM"),
    dir = "microbiome_lda_models/",
    reset = TRUE,
    verbose = TRUE
  )

```

We can now align the topics from each consecutive models:


```{r align-topics}


aligned_topics_product <- 
  align_topics(
    models = lda_models,
    method = "product") # transport product

aligned_topics_transport <- 
  align_topics(
    models = lda_models,
    method = "transport") 


aligned_topics_product@weights %>% head()



```

And visualize the number of key topics, the alignments colored by refinement scores, stability scores or key topics, and the topic composition for specific models.

```{r align-topics-viz}

g_key_topics_product <- compute_number_of_key_topics(aligned_topics_product) %>% plot_number_of_key_topics() + ggtitle("Method: product")
g_key_topics_transport <- compute_number_of_key_topics(aligned_topics_transport) %>% plot_number_of_key_topics() + ggtitle("Method: transport")


g_branches_product <- plot_alignment(aligned_topics_product)
g_branches_transport <- plot_alignment(aligned_topics_transport)

g_refinement_product <- plot_alignment(aligned_topics_product, color_by = "re")
g_refinement_transport <- plot_alignment(aligned_topics_transport, color_by = "re")


g_robustness_product <- plot_alignment(aligned_topics_product, color_by = "ro")
g_robustness_transport <- plot_alignment(aligned_topics_transport, color_by = "ro")

g_beta_product <-  plot_beta(aligned_topics_product, models = c("k3", "k6","k12","k20"), min_beta = 0.005)
g_beta_transport <-  plot_beta(aligned_topics_transport, models = c("k3", "k6","k12","k20"), min_beta = 0.005)


figure = ggpubr::ggarrange(
  g_key_topics_product,
  g_key_topics_transport,
  g_refinement_product,
  g_refinement_transport,
  g_robustness_product, 
  g_robustness_transport,
  g_branches_product,
  g_branches_transport,
  g_beta_product,
  g_beta_transport,
  ncol = 2,
  nrow = 5,
  heights = c(1.5,2,2,2,3)
)

ggsave(figure, filename = "microbiome_figure.png", height = 22, width = 18, units = "cm", scale = 2)

```



## Key number of topics

```{r key-number-of-topics}



```


## Alignment with diagnostics scores


```{r}

plot_alignment(aligned_topics_product, color_by = "re")
plot_alignment(aligned_topics_transport, color_by = "re")

```





```{r}


plot_alignment(aligned_topics_product, color_by = "ro")
plot_alignment(aligned_topics_transport, color_by = "ro")

```

```{r}

plot_beta(aligned_topics_product, models = c("k3","k3","k12", "k20"), min_beta = 0.005, color_by  = "branch")
plot_beta(aligned_topics_product, models = c("k3","k6","k12", "k20"), min_beta = 0.005, color_by  = "branch")

plot_beta(aligned_topics_product, models = c("k3","k6","k12", "k20"), min_beta = 0.005, color_by  = "robustness")

```







