---
title: "microbiome-demo"
output: rmarkdown::html_vignette
vignette: >
%\VignetteIndexEntry{microbiome-demo}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
  ---
  
  ```{r, include = FALSE}
knitr::opts_chunk$set(
collapse = TRUE,
comment = "#>",
eval = FALSE
)
```

```{r setup}
library(alto)
library(dplyr)
library(magrittr)
library(ggplot2)
library(tibble)
library(stringr)
library(tidyr)
```


This document showcases the `alto` package functions applied to vaginal microbiome data.

## Loading the data

We use the data published with the article ["Replication and refinement of a vaginal microbial signature of preterm birth in two racially distinct cohorts of US women"](https://www.pnas.org/content/114/37/9966.short) by Calahan et al., 2017, PNAS.

A .zip file containing the related data can be downloaded from the [Stanford data catalog](https://stacks.stanford.edu/file/druid:yb681vm1809/RepRefine_Scripts.tar.gz).

For convenience, the data file (i.e. the tables contained in the `processed.rda` file) has been attached to the `alto` package as `vm_data`. For details, type `?vm_data`.

```{r show-names-of-provided-data}

names(vm_data)

```

First, we do some pre-processing of the data. Specifically, we change the `colnames` to replace ASV sequences by human-friendly names built from the taxonomy table.

```{r human-friendly-ASV-names}

# we create the human-friendly ASV names
# by concatenating Genus and Species and an identifying number
# as a given species can be represented by several ASVs
vm_data$taxonomy <-
  vm_data$taxonomy %>%
  as.data.frame() %>%
  group_by(Genus, Species) %>%
  mutate(
    ASV_name =
      paste0(
        ifelse(is.na(Genus),"[unknown Genus]", Genus), " (",
        ifelse(is.na(Species),"?", Species), ") ",
        row_number()
      )
  ) %>%
  ungroup()

# then, we replace the colnames of the count matrix by these new names.
# Note that the tax table rows are ordered as the count matrix columns,
# allowing us to assign without matching first.
colnames(vm_data$counts) <-  vm_data$taxonomy$ASV_name

```



## Using `alto` on these data

Our first step is to run the lda models for varying number of topics, i.e. from 2 to 15 topics.
For this, we use the `run_lda_models` function. This can take a while on a personal computer.
For example, it takes about x minutes on the authors' laptops.

```{r run-lda-models}
lda_varying_params_lists <-  list()
for (k in 1:20) {lda_varying_params_lists[[paste0("k",k)]] <- list(k = k)}

lda_models  <- 
  run_lda_models(
    data = vm_data$counts, #[,1:100], # change for the full data set when full workflow is ready
    lda_varying_params_lists = lda_varying_params_lists,
    lda_fixed_params_list = list(method = "VEM"),
    dir = "microbiome_lda_models/",
    reset = FALSE,
    verbose = TRUE
  )

```

We can now align the topics from each consecutive models:


```{r align-topics}


aligned_topics_product <- 
  align_topics(
    models = lda_models,
    method = "product") # transport product

aligned_topics_transport <- 
  align_topics(
    models = lda_models,
    method = "transport") 


aligned_topics_product@weights %>% head()
<<<<<<< Updated upstream


=======
>>>>>>> Stashed changes

```

## Alignement viz


```{r alignment-viz}


g_branches_product <- plot_alignment(aligned_topics_product)
g_branches_product
g_branches_transport <- plot_alignment(aligned_topics_transport)
g_branches_transport


```


<<<<<<< Updated upstream
g_beta_product <-  plot_beta(aligned_topics_product, models = c("k3", "k6","k12","k20"), min_beta = 0.005)
g_beta_transport <-  plot_beta(aligned_topics_transport, models = c("k3", "k6","k12","k20"), min_beta = 0.005)
=======
>>>>>>> Stashed changes

## Key number of topics

```{r key-number-of-topics}

g_key_topics_product <- compute_number_of_key_topics(aligned_topics_product) %>% plot_number_of_key_topics() + ggtitle("Method: product")
g_key_topics_product
g_key_topics_transport <- compute_number_of_key_topics(aligned_topics_transport) %>% plot_number_of_key_topics() + ggtitle("Method: transport")
g_key_topics_transport



```


## Alignment with diagnostics scores



```{r refinement}


g_refinement_product <- plot_alignment(aligned_topics_product, color_by = "re")
g_refinement_product
g_refinement_transport <- plot_alignment(aligned_topics_transport, color_by = "re")
g_refinement_transport

```





<<<<<<< Updated upstream
plot_alignment(aligned_topics_product, color_by = "re")
plot_alignment(aligned_topics_transport, color_by = "re")
=======
```{r robustness}

g_robustness_product <- plot_alignment(aligned_topics_product, color_by = "ro")
g_robustness_product
g_robustness_transport <- plot_alignment(aligned_topics_transport, color_by = "ro")
g_robustness_transport
>>>>>>> Stashed changes

```

```{r}


g_beta_product <-  plot_beta_lsy(aligned_topics_product, models = c("k3", "k6","k12","k20"), min_beta = 0.005)
g_beta_product
g_beta_transport <-  plot_beta_lsy(aligned_topics_transport, models = c("k3", "k6","k12","k20"), min_beta = 0.005)
g_beta_transport


```
```{r}


<<<<<<< Updated upstream
plot_alignment(aligned_topics_product, color_by = "ro")
plot_alignment(aligned_topics_transport, color_by = "ro")
=======
g_beta_product_ro <-  plot_beta_lsy(aligned_topics_product, models = c("k3", "k6","k12","k20"), min_beta = 0.005, color_by = "ro")
g_beta_product_ro


#plot_beta_lsy(aligned_topics_product, models = c("k12"), min_beta = 0.005, color_by = "ro")

>>>>>>> Stashed changes

```


<<<<<<< Updated upstream
plot_beta(aligned_topics_product, models = c("k3","k3","k12", "k20"), min_beta = 0.005, color_by  = "branch")
plot_beta(aligned_topics_product, models = c("k3","k6","k12", "k20"), min_beta = 0.005, color_by  = "branch")

plot_beta(aligned_topics_product, models = c("k3","k6","k12", "k20"), min_beta = 0.005, color_by  = "robustness")
=======

>>>>>>> Stashed changes



```{r align-topics-viz-figure, echo = FALSE, eval = FALSE}


figure_full = ggpubr::ggarrange(
  g_key_topics_product,
  g_key_topics_transport,
  g_refinement_product,
  g_refinement_transport,
  g_robustness_product, 
  g_robustness_transport,
  g_branches_product,
  g_branches_transport,
  g_beta_product,
  g_beta_transport,
  ncol = 2,
  nrow = 5,
  heights = c(1.5,2,2,2,3)
)

ggsave(figure_full, filename = "microbiome_figure.png", height = 22, width = 18, units = "cm", scale = 2)


```




```{r align-topics-viz-figure-2, echo = FALSE, eval = FALSE}


make_figure <- function(aligned_topics_product,
                        aligned_topics_transport){
  
  
  n_key_topics_combined <- 
    bind_rows(
      compute_number_of_key_topics(aligned_topics_product) %>% 
        mutate(method = "product"),
      compute_number_of_key_topics(aligned_topics_transport) %>% 
        mutate(method = "transport")
    )
  
  
  g_key_topics <- 
    ggplot(n_key_topics_combined, 
           aes(x = n_topics,
               y = n_key_topics,
               col = method)) +
    geom_abline(slope = 1, intercept = 0, col = "gray90", size = 2) +
    geom_line() +
    geom_point(size = 1) +
    theme_minimal() +
    xlab("# of topics in model") +
    ylab("# of key topics") +
    scale_x_continuous(breaks = 1:max(n_key_topics_combined$n_topics), minor_breaks = NULL) +
    scale_y_continuous(breaks = 1:max(n_key_topics_combined$n_topics), minor_breaks = NULL) +
    theme(legend.position = "top")
  
  
  g_robustness_transport <- 
    plot_alignment(
      aligned_topics_transport, 
      color_by = "ro", 
      model_name_repair_fun = function(x) str_remove_all(x, "k")
    ) + 
    theme_minimal() +
    theme(legend.position = "left")
  
  
  g_branches_transport <- 
    plot_alignment(
      aligned_topics_transport, 
      model_name_repair_fun = function(x) str_remove_all(x, "k")
    ) +
    theme_minimal() 
  
  
  g_beta_transport =  plot_beta_lsy(aligned_topics_transport, models = c("k3", "k7","k12",last(aligned_topics_transport@topics$m %>% levels())), min_beta = 0.003)
  
  
  
  figure = 
    ggpubr::ggarrange(
      ggpubr::ggarrange(
        g_key_topics,
        g_robustness_transport,
        g_branches_transport,
        ncol = 3, nrow = 1,
        widths = c(3,5,4),
        labels = "auto"),
      g_beta_transport,
      ncol = 1,
      nrow = 2,
      labels = c("","d")
    )
  
  figure
  
  
}

aligned_topics_product_18 <- align_topics(lda_models[1:18], method = "product")
aligned_topics_transport_18 <- align_topics(lda_models[1:18], method = "transport")


figure_1_18 <- make_figure(aligned_topics_product = aligned_topics_product_18, 
                           aligned_topics_transport = aligned_topics_transport_18)


ggsave(figure_1_18, filename = "microbiome_figure.png", height = 14, width = 18, units = "cm", scale = 2)


```
